capstone/src/main/webapp/script.js

```diff
@@ -1,31 +1,153 @@
 function createMap() {
-    const map = new google.maps.Map(
-        document.getElementById('map-container'),
-        {center: {lat: 37.7749, lng: -122.4194}, zoom: 12}
-    );
-  
-    //**adding zipcode overlay */
-    map.data.loadGeoJson('zipcode-data.json');
-    //**adding precinct overlay */
-    map.data.loadGeoJson('neighborhoods.json');
-    
-    map.data.setStyle({visible: false});
-
-    const cityBorder = [
-        {lat: 37.708305, lng: -122.502691},
-        {lat: 37.708229, lng: -122.393322}
-    ]
-
-    const cityLimit = new google.maps.Polygon({
-        paths: cityBorder,
-        strokeColor: 'black',
-        strokeOpacity: 0.8,
-        strokeWeight: 2,
-        fillColor: 'black',
-        fillOpacity: 0.35
-    });
-  
+  const map = new google.maps.Map(document.getElementById('map-container'),
+      {
+        center: {lat: 37.7749, lng: -122.4194},
+        zoom: 12,
+      });
+
+  const cityBorder = [
+    {lat: 37.708305, lng: -122.502691},
+    {lat: 37.708229, lng: -122.393322},
+  ];
+
+  const cityLimit = new google.maps.Polygon({
+    paths: cityBorder,
+    strokeColor: 'black',
+    strokeOpacity: 0.8,
+    strokeWeight: 2,
+    fillColor: 'black',
+    fillOpacity: 0.35,
+  });
+
   cityLimit.setMap(map);
+
+  //* *button for centering map*/
+  const centerControlDiv = document.createElement('div');
+  const centerControl = new CenterControl(centerControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER].push(centerControlDiv);
+
+  //* *button for zipcode data layer */
+  const zipControlDiv = document.createElement('div');
+  const zipControl = new ZipControl(zipControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER].push(zipControlDiv);
+
+  //* *button for zipcode data layer */
+  const precinctControlDiv = document.createElement('div');
+  const precinctControl = new PrecinctControl(precinctControlDiv, map);
+  map.controls[google.maps.ControlPosition.LEFT_CENTER]
+      .push(precinctControlDiv);
+}
+
+function CenterControl(controlDiv, map) {
+  //* *button creation and positioning*/
+  const controlUI = document.createElement('div');
```

Comment:
Can we put this div in the html as an empty div and then we can move all the styling to style.css to clean up this file

Reply:
I don't think that will clean up the file, and instead spread out my work unnecessarily. I'll keep this in mind for next time though!

Reply:
![Screenshot 2020-07-06 at 5 52 48 PM](https://user-images.githubusercontent.com/16478651/86829769-6e662900-c05a-11ea-8407-524584953de8.png), the buttons won't be where I need them to be if I move the html and in the tutorial (https://developers.google.com/maps/documentation/javascript/examples/control-custom#all) that's where it is.